### PR TITLE
Feature/02 data fetch init

### PR DESCRIPTION
**Pull request by luszczka**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # dependencies
 /node_modules
 /.next
-/.husky
\ No newline at end of file
+/.husky
+
+# local env files
+.env*.local
```
Comment (misiq) [inline .gitignore:7]: hmmm this will be working for .env and .env-local? how?

File: package.json [hunk 2]
```diff
@@ -1,8 +1,13 @@
 {
+  "name": "drink-me",
+  "version": "1.0.2",
   "scripts": {
     "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
     "lint": "eslint .",
     "lint:fix": "eslint . --fix",
+    "push": "yarn version --patch && git push --follow-tags",
```
Comment (misiq) [inline package.json]: ?

File: src/hooks/useDrinkDetails.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+const useDrinkDetails = (id: string): DrinkDetails | undefined => {
+  const [data, setData] = useState<DrinkDetails>();
+
+  useEffect(() => {
+    const fetchDrinkDetails = async (): Promise<void> => {
+      try {
+        const result = await fetch(`${process.env.NEXT_PUBLIC_URL}/lookup.php?i=${id}`);
+        const response = await result.json();
+        setData(response.drinks[0]);
```
Comment (misiq) [inline src/hooks/useDrinkDetails.ts]: what if you get ann empty array In drinks

File: .gitignore (hunk 1 above)
Comment (luszczka) [reply .gitignore:7]: my env local file is env.local so it works fine

File: src/hooks/useRandomDrink.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { type RandomDrink } from '../utils/APIResponsesTypes';
+import { useEffect, useState } from 'react';
+
+const useRandomDrink = (): RandomDrink | undefined => {
+  const [data, setData] = useState<RandomDrink>();
+
+  useEffect(() => {
+    const fetchRandomDrink = async (): Promise<void> => {
```
Comment (misiq) [inline src/hooks/useRandomDrink.ts]: you use this [...]  const result = await fetch(`${process.env.NEXT_PUBLIC_URL}/lookup.php?i=${id}`);
        const response = await result.json(); [...] multiple times maybe create hook useFetch?

File: package.json (hunk 2 above)
Comment (luszczka) [reply package.json]: this increments my project version on push

File: src/hooks/useSearchDrinkBy.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { type DrinkByIngredient, type DrinksByName } from '../utils/APIResponsesTypes';
+import { useEffect, useState } from 'react';
+
+export enum SearchType {
```
Comment (misiq) [inline src/hooks/useSearchDrinkBy.ts]: s & i why?

File: src/utils/APIResponsesTypes.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  strDrink: string;
+  strDrinkAlternate: string;
+  strDrinkThumb: string;
+  strGlass: string;
+  strIBA: string;
+  strImageAttribution: string;
+  strImageSource: string;
+  strIngredient1: string;
+  strIngredient2: string;
+  strIngredient3: string;
```
Comment (misiq) [inline src/utils/APIResponsesTypes.ts:16]: Do you know you can generate this from API?

**Review by misiq — CHANGES_REQUESTED** [review]
Few things to resolve

File: package.json (hunk 2 above)
Comment (misiq) [reply package.json]: I know, but do you want to increment the version on any push? 

IMO you should increment when you change something, sometime you will push it to repo but your feature will not be completed 

MAJOR version when you make incompatible API changes
MINOR version when you add functionality in a backward-compatible manner
PATCH version when you make backward-compatible bug fixes

File: src/components/Container/Container.styled.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import styled from 'styled-components';
+
+export const StyledContainer = styled.div`
+  width: 100vw;
+  margin: 40px;
```
Comment (misiq) [inline src/components/Container/Container.styled.ts:5]: it's ok, but there is an approach to not use margin on elements but add a gap on the parent. to take this in mind in the future.

File: src/components/DrinkOfTheDay/DrinkOfTheDay.tsx [hunk 8]
```diff
@@ -1,12 +1,28 @@
...
-import { StyledDrinkOfTheDayHeader, StyledDrinkOfTheDayWrapper } from './DrinkOfTheDay.styled'
+import { type ReactElement } from 'react';
+import useFetch from '../../hooks/useFetch';
+import { type RandomDrink } from '../../utils/APIResponsesTypes';
+import SingleBox from '../SingleBox/SingleBox';
+import { StyledDrinkOfTheDayHeader, StyledDrinkOfTheDayWrapper } from './DrinkOfTheDay.styled';
 
 const DrinkOfTheDay = (): ReactElement => {
+  const { data } = useFetch<{ drinks: RandomDrink[] }>('/random.php') || {};
+  const { drinks } = data ?? {};
```
Comment (misiq) [inline src/components/DrinkOfTheDay/DrinkOfTheDay.tsx]: if you move this after if you will not need this ?? {}

File: src/pages/dashboard/index.tsx [hunk 9]
```diff
@@ -1,25 +1,27 @@
-import { type ReactElement } from 'react';
+import { useEffect, useRef, useState, type ReactElement } from 'react';
 import DrinkOfTheDay from '../../components/DrinkOfTheDay/DrinkOfTheDay';
+import InputSearch from '../../components/InputSearch/InputSearch';
 import Navigation from '../../components/Navigation/Navigation';
 import PageWrapper from '../../components/PageWrapper/PageWrapper';
+import useFetch from '../../hooks/useFetch';
+import { type DrinksByName } from '../../utils/APIResponsesTypes';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const debounce = require('lodash.debounce');
```
Comment (misiq) [inline src/pages/dashboard/index.tsx]: import { debounce } from "lodash"

File: package.json [hunk 10]
```diff
@@ -11,6 +15,8 @@
     "@mui/icons-material": "^5.10.9",
     "@mui/material": "^5.10.13",
     "@mui/styled-engine-sc": "^5.10.6",
+    "@tanstack/react-query": "^4.24.4",
+    "lodash.debounce": "^4.0.8",
```
Comment (misiq) [inline package.json:19]: npm install @types/lodash

**Review by misiq — CHANGES_REQUESTED** [review]
Keep up the good work!

**Pull request merged**